src/plugins/user-activity-events/client.js

```diff
@@ -0,0 +1,34 @@
+/* eslint-disable no-invalid-this */
+'use strict';
+
+function throttle(fn) {
+	var last, deferTimer;
+	return function() {
+		var context = this;
+		//only fire the event if it has been at least 10 seconds since previous fire
+		var threshhold = 10000;
+		var now = +new Date();
+		if (last && now < last + threshhold) {
+			// hold on to it
+			clearTimeout(deferTimer);
+			deferTimer = setTimeout(function() {
+				last = now;
+				fn.apply(context);
+			}, threshhold + last - now);
+		} else {
+			last = now;
+			fn.apply(context);
+		}

```

Comment:
I like this logic a lot -- it ends up being super responsive to user interactions but doesn't flood the postMessage pipe. 👍